src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -664,6 +661,23 @@ ucs_status_t uct_dc_mlx5_ep_flush(uct_ep_h tl_ep, unsigned flags,
                                       comp, sn);
 }
 
+ucs_status_t uct_dc_mlx5_ep_invalidate(uct_ep_h tl_ep, unsigned flags)
+{
+    uct_dc_mlx5_ep_t    *ep    = ucs_derived_of(tl_ep, uct_dc_mlx5_ep_t);
+    uct_dc_mlx5_iface_t *iface = ucs_derived_of(tl_ep->iface, uct_dc_mlx5_iface_t);
+    uct_ib_mlx5_md_t    *md    = ucs_derived_of(iface->super.super.super.super.md,
+                                                uct_ib_mlx5_md_t);
+    UCT_DC_MLX5_TXQP_DECL(txqp, txwq);
+
+    ep->flags |= UCT_DC_MLX5_EP_FLAG_INVALID;
```

Comment:
i think setting it before checking if have DCI will cause modify to err state twice

Reply:
I think it can be moved under `if (ep->dci == UCT_DC_MLX5_EP_NO_DCI)` if add NOP